### PR TITLE
Fixed non-symplecticity bug in LinearTransferMatrix

### PR DESCRIPTION
**Pull request by pkruyt**
Reopening #271

### CODE REVIEW

File: xtrack/beam_elements/elements_src/lineartransfermatrix.h [hunk 1]
```diff
@@ -197,13 +205,12 @@ void LinearTransferMatrix_track_local_particle(LinearTransferMatrixData el, Loca
...
-        
     // re-adding dispersion and closed orbit
     delta = LocalParticle_get_delta(part);
+    rvv = LocalParticle_get_rvv(part);
+    LocalParticle_add_to_zeta(part,(-disp_px_1*LocalParticle_get_x(part) + disp_x_1*LocalParticle_get_px(part) - disp_py_1*LocalParticle_get_y(part) + disp_y_1*LocalParticle_get_py(part))/rvv);
     LocalParticle_add_to_x(part,disp_x_1 * delta + x_ref_1);
-    LocalParticle_add_to_px(part,px_ref_1);
+    LocalParticle_add_to_px(part,px_ref_1 +disp_px_1 * delta);
     LocalParticle_add_to_y(part,disp_y_1 * delta + y_ref_1);
     LocalParticle_add_to_py(part,py_ref_1);
```
Comment (giadarol) [inline xtrack/beam_elements/elements_src/lineartransfermatrix.h:215]: ```+disp_py_1 * delta``` seems to be missing here

**Comment by giadarol** [comment]
I was making some tests. Apparently the correction on zeta that is supposed to make the map symplectic in fact breaks symplecticity (I find that the determinant of the jacobian of the transformation is 0.8!).

Here an example showing the problem:
https://github.com/xsuite/xtrack/blob/issue_with_simplectified_linear_transf_matrix/examples/transfer_matrix/002_check_with_twiss.py

The matrix is much better behaved if I comment out the correction on zeta.

**Comment by giadarol** [comment]
I am testing with more effects included (synchrotron motion, non zero orbit) and for an energy of 1 GeV. Could you please repeat your check for the parameters in this example:
https://github.com/xsuite/xtrack/blob/issue_with_simplectified_linear_transf_matrix/examples/transfer_matrix/002_check_with_twiss.py

File: ducktrack/elements.py [hunk 2]
```diff
@@ -755,10 +765,17 @@ def track(self,p):
             p.delta += self.gauss_noise_ampl_s*np.random.randn(len(p.delta))
 
         # re-adding dispersion and closed orbit
+        old_x=p.x
+        old_px=p.px
+        old_y=p.y
+        old_py=p.py
+
```
Comment (giadarol) [inline ducktrack/elements.py:772]: This is not a copy but just a reference.

To have a full copy of the array you need:
```python
old_x = p.x.copy()
```

Comment (pkruyt) [reply ducktrack/elements.py]: Fixed

**Comment by pkruyt** [comment]
> I am testing with more effects included (synchrotron motion, non zero orbit) and for an energy of 1 GeV. Could you please repeat your check for the parameters in this example: https://github.com/xsuite/xtrack/blob/issue_with_simplectified_linear_transf_matrix/examples/transfer_matrix/002_check_with_twiss.py

Sorry for the delay. I get the following results:

With zeta correction:

determinant         : 0.9999959105499323
symplectic check : 2.828299971438128

Without zeta correction: 

determinant         : 0.9999953777998484
symplectic check : 6.647808767832795

File: xtrack/beam_elements/elements_src/lineartransfermatrix.h (hunk 1 above)
Comment (pkruyt) [reply xtrack/beam_elements/elements_src/lineartransfermatrix.h]: Fixed

**Comment by pkruyt** [comment]
I also added the disp_py and now the performance is a lot better:

With correction: 
symplectic check: 0.0196

Without correction:
symplectic check: 6.465

**Comment by giadarol** [comment]
This has now diverged w.r.t the main branch. I have opened an issue for this item (https://github.com/xsuite/xsuite/issues/354). Closing this PR.

**Pull request closed without merging**